compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2277,4 +2269,18 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
       }
     }
   }
+
+  protected def checkEqualityEvidence(tree: tpd.Tree, pt: Type)(implicit ctx: Context) : Unit = {
```

Comment:
/** Check that `tree == x: pt` is typeable. Used when checking a pattern
     *  againts a selector of type `pt`. This implementation accounts for
     *  user-defined definitions of `==`.
     *  Overwritten to no-op in ReTyper.
     */